apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/_components/DocumentsLayout/index.tsx

```diff
@@ -28,10 +31,22 @@ export default async function DocumentsLayout({
   commitUuid: string
 }) {
   const session = await getCurrentUser()
-  const project = await findProjectCached({
-    projectId,
-    workspaceId: session.workspace.id,
-  })
+  let project
+  try {
+    project = await findProjectCached({
+      projectId,
+      workspaceId: session.workspace.id,
+    })
+  } catch (error) {
+    console.warn((error as Error).message)
+
+    if (error instanceof NotFoundError) {
+      return redirect(ROUTES.dashboard.root)
+    }
+
+    throw error
```

Comment:
We need an abstraction here. By abstraction undertsand how to handle 404 in nextjs propoerly without needing this try/catch